atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,228 @@
+            throws InterruptedException {
+        TodoResource clientToSingleNode = createClientFor(TodoResource.class, asPort("ete1"));
+
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+
+        String cassandraNodeToKill = getRandomCassandraNodeToShutdown();
+        killCassandraContainer(cassandraNodeToKill);
+
+        long transactionStartTime = System.currentTimeMillis();
+        assertAddTodoTransactionWasSuccessful(clientToSingleNode);
+        long transactionEndTime = System.currentTimeMillis();
+
+        long transactionTimeAfterNodeIsKilled = transactionEndTime - transactionStartTime;
+
+        startCassandraContainer(cassandraNodeToKill);
+
+        assertThat("transactionTimeAfterNodeIsKilled",
+                transactionTimeAfterNodeIsKilled,
+                is(lessThan(MAX_CASSANDRA_NODE_DOWN_MILLIS)));
+    }
+
+    private static DockerPort asPort(String node) {
+        return MULTINODE_CASSANDRA_SETUP.containers().container(node).port(TIMELOCK_SERVER_PORT);
+    }
+
+    private static <T> T createClientFor(Class<T> clazz, DockerPort port) {
+        String uri = port.inFormat("http://$HOST:$EXTERNAL_PORT");
+        return AtlasDbHttpClients.createProxy(Optional.absent(), uri, clazz);
+    }
+
+    private static HealthCheck<Container> toBeReady() {
+        return (container) -> {
+            TodoResource todos = createClientFor(TodoResource.class, container.port(TIMELOCK_SERVER_PORT));
+
+            return SuccessOrFailure.onResultOf(() -> {
+                todos.isHealthy();
+                return true;
+            });
+        };
+    }
+
+    private String getRandomCassandraNodeToShutdown() {
+        int index = ThreadLocalRandom.current().nextInt(CASSANDRA_NODES.size());
+        return CASSANDRA_NODES.get(index);
+    }
+
+    private static void killCassandraContainer(String containerName) {
+        Container container = MULTINODE_CASSANDRA_SETUP.containers().container(containerName);
+        try {
+            container.kill();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void startCassandraContainer(String containerName) throws InterruptedException {
+        Container container = MULTINODE_CASSANDRA_SETUP.containers().container(containerName);
+        try {
+            container.start();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        waitForAllPorts(container);
+        waitForNodetoolToConfirmStatus(container, "UN", 3);
+    }
+
+    private static void waitForAllPorts(Container container) {
+        Awaitility.await()
+                .atMost(50, TimeUnit.SECONDS)
+                .until(() -> container.areAllPortsOpen().succeeded());
+    }
+
+    private void assertAddTodoTransactionWasSuccessful(TodoResource todoClient) {
```

Comment:
Proposed solution: extract the logic into something like "TodoEteTestUtils". What do you think?